feat(grouping): Document hierarchical grouping (#3916)
getsentry/sentry-docs: src/docs/product/data-management-settings/event-grouping/grouping-breakdown/index.mdx

If you’re interested in being an Early Adopter, you can turn your organization’s Early Adopter status on/off in General Settings. This will affect all users in your organization and can be turned back off just as easily.

</Note>

Sentry's current grouping algorithm puts two events into the same issue if they have the same stack trace. This approach sometimes creates multiple distinct issues for errors with the same root cause.

If this happens occasionally, you might still be able reduce the noise by manually [merging](/product/issues/issue-details/#tabs) those issues. If, however, you want coarser issue groups by default, follow the steps below.

## Enabling the grouping breakdown (beta)

These steps will enable our newest grouping algorithm, which creates coarser
(that is, fewer) issues by considering only the most relevant part of the stack trace.

1. Join our Early Adopter program _(Settings &raquo; General Settings)_.

1. [Upgrade to the latest grouping algorithm](/product/data-management-settings/event-grouping#grouping-algorithms).

<Alert level="warning">
Note that upgrading the grouping algorithm is an irreversible step.
</Alert>

> ## Using the grouping breakdown

Author: The grouping breakdown is only available after updating the grouping algorithm. So should we really write _updated_ grouping breakdown here?

> ## Using the grouping breakdown

Reviewer: The user flow, if I understand correctly, is that the first step is to enable the new grouping, the next step is to use it. Thus, "updated" makes sense here.

If you have followed the steps described above, the Grouping Breakdown allows you to navigate subgroups of an issue, to see what events would group together if a larger
part of the stack trace were to be considered.

Take, for example, a crashing function which is called from multiple
locations in your code ("intermediate function" "1" and "2"). When you move the slider
all the way to the left, only the crashing frame is considered for grouping, and
all events are sorted into the same issue regardless of the calling location:

![Breakdown of this issue on the highest level](breakdown_level_01.png)

When you move the slider to the right, you can see what groups would be created
if the calling frame were also to be considered:

![Breakdown of this issue on an intermediate level](breakdown_level_02.png)

You can add another level by moving the slider all the way to the right.
This, however, does not add any new subgroups, as both calling functions are themselves
called from the same location (`main`):

![Breakdown of this issue on the deepest level](breakdown_level_03.png)

> TODO: retina quality screenshots?

Reviewer 2: @jjbayer yes, I would do this. you can probably screenshot on the macos internal screen to achieve, or zoom in on the website

> See [Sentinel & Prefix Frames](/product/data-management-settings/event-grouping/stack-trace-rules#sentinel--prefix-frames) on how to change which parts of the stack trace are used for grouping.

Reviewer: ```suggestion
[Sentinel & Prefix Frames](/product/data-management-settings/event-grouping/stack-trace-rules#sentinel--prefix-frames) discusses how to change the parts of the stack trace used for grouping.
```
